### DISCUSSION TITLE
Brackets, Braces and Parentheses - leveling the playing field.

### DISCUSSION THREAD

**Discussion by fade2gray** [mmikeww/AHK-v2-script-converter#227] [Ideas]
Firstly, sorry about the novice PR.

I asked ChatGPT, "What would you collectively call these characters, [ ] { } ( )".

The reply ...
> These characters are collectively called brackets. They can be further categorized into specific types:
> 
>     Square brackets or box brackets: [ ]
>     Curly brackets or braces: { }
>     Parentheses or round brackets: ( )
> 
> In formal or technical contexts, referring to them simply as "brackets" is often understood, but specifying the type (square, curly, or parentheses) can provide additional clarity.

I propose, for the sake of clarity within the code, that "bracket/s" is used for the collective term, and the terms `bBracket` (box bracket), `cBracket` (curly bracket) and `rBracket` (round bracket) are used where required.

For example, in `QuickConvertorV2_scintilla.ahk`, `Brace` and `brace/s` would be replaced with `Bracket` and `bracket/s` respectively.

In `ConverFuncs.ahk` ...

`try code := AddBracket(code)         ; Add Brackets to Hotkeys`
becomes 
`try code := Add_cBracket(code)         ; Add cBrackets to Hotkeys`

```Result .= LabelPointer = 1 ? "} `; V1toV2: Added bracket before function`r`n" : ""```
becomes
```Result .= LabelPointer = 1 ? "} `; V1toV2: Added closing cBracket`r`n" : ""```
etc,
etc,

I also propose, that **all** comments added to the **converted** code should be prefixed with `; V1toV2:`, such as ...
`Return "; REMOVED Random reseed"` => `Return "; V1toV2: REMOVED Random reseed"`
... this would make it a lot easier for the end user to search for and remove any comments not pertaining to their code.

Am I going too far?

**Comment by mmikeww** [comment]
i use brackets, braces, parenthesis respectively, i think a new reader of the code would be able to understand what those mean easier than someone seeing cBracket

**Comment by andymbody** [comment]
I like the idea of the consistent comment headers added to converted code. It would be it a cinch to find/replace, especially with regex in Notepad++. I have no preference regarding the term for brackets. Can just call them brackets as far as I'm concerned (rather than tagging them with b,c,r). I doubt it will make much difference to those reading the code. Just my opinion...

**Reply by Banaanae** [reply]
> Can just call them brackets as far as I'm concerned

I too am guilty of this, I don't think it's too bad as the comments always show before the bracket in question
`} ; V1toV2: Added Bracket before label`

> I also propose, that all comments added to the converted code should be prefixed with `; V1toV2:`

I too want this (I have been retroactively changing them as I find them, probably should do mass find replace

**Comment by Banaanae** [comment]
I agree with your comment prefixes, but I would like to keep the `; REMOVED: ` (with colon) prefix too

This way `; V1toV2:` shows what changes need to be made that the converter can't do or useful notes with the conversion
and `; REMOVED: ` shows what has been removed from the source script (usually from being removed in v2)

**Comment by fade2gray** [comment]
@Banaanae don't understand why you are ok with `; V1toV2: Added ...`, but not `; V1toV2: Removed ...`.
Would `; V1toV2: ADDED ...` and `V1toV2: REMOVED ...` be more acceptable?

I just thought it would be easier for the end user to search for a single prefix, "V1toV2", in order to aid the removal of all comments added by the converter.

All comments added by the converter can be removed by using two regex's, `^; V1toV2:.*\r?\n` and ` ; V1toV2:.*`, which gives me another idea, add an option to the Gui to auto remove the added comments once the resulting code has been tested as successful - just a thought.

**Reply by Banaanae** [reply]
Putting it that way it makes much more sense, I understand now :)

**Reply by Banaanae** [reply]
All comment should now have `; V1toV2: ` in #257

**Comment by JoeWinograd** [comment]
Yes, too far, imo. I think that the bulk of the programming/scripting world uses braces, brackets, and parentheses without confusion. If you want to go one step further, maybe square brackets and/or curly braces. I don't recollect ever hearing box brackets or round brackets. The latter for parentheses is very strange...re ChatGPT on that one...too much A, not enough I. :) Regards, Joe

P.S. I replied to this via email on 2-Jul, but never got a notification email back with it, so I presume it wasn't delivered. I just discovered the Discussions tab on GitHub, so I'm trying again here. This is my first ever GitHub contribution.
